docs/architecture/adr-065-custom-event-indexing.md

```diff
@@ -0,0 +1,341 @@
+# ADR 065: Custom Event Indexing
+
+- [ADR 065: Custom Event Indexing](#adr-065-custom-event-indexing)
+  - [Changelog](#changelog)
+  - [Status](#status)
+  - [Context](#context)
+  - [Alternative Approaches](#alternative-approaches)
+  - [Decision](#decision)
+  - [Detailed Design](#detailed-design)
+    - [EventSink](#eventsink)
+    - [Supported Sinks](#supported-sinks)
+      - [`KVEventSink`](#kveventsink)
+      - [`PSQLEventSink`](#psqleventsink)
+    - [Configuration](#configuration)
+  - [Consequences](#consequences)
+    - [Positive](#positive)
+    - [Negative](#negative)
+    - [Neutral](#neutral)
+  - [References](#references)
+
+## Changelog
+
+- April 1, 2021: Initial Draft (@alexanderbez)
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, Tendermint Core supports block and transaction event indexing through
+the `tx_index.indexer` configuration. Events are captured in transactions and
+are indexed via a `TxIndexer` type. Events are captured in blocks, specifically
+from `BeginBlock` and `EndBlock` application responses, and are indexed via a
+`BlockIndexer` type. Both of these types are managed by a single `IndexerService`
+which is responsibly for consuming events and sending those events off to be
+indexed by the respective type.
+
+In addition to indexing, Tendermint Core also supports the ability to query for
```

Comment:
This did facilitate clients for a while, but has hit it's limits some time ago. Many block explorers, like Big Dipper, (and even wallets like  Lunie) pull the data (using custom techniques) into a custom database in order to provide nice queries for the UI.

We keep using this interface for apps as well, but there are plenty of limitations and work-arounds, and I understand the system quite well.

I think we should aim to **replace** the tendermint indexer service with a pluggable data sink, each of which can offer new APIs. And ideally sponsor a few variants via the ICF that can be forked and improved